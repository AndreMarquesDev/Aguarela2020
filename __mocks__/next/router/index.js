jest.mock('next/router', () => ({
    ...jest.requireActual('next/router'),
    useRouter() {
        return {
            route: '/',
            pathname: '',
            query: {
                language: 'pt',
            },
            asPath: '',
        };
    },
}));

module.exports = jest.requireMock('next/router');
