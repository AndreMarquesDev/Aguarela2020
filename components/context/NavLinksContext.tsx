import { createContext, Context } from 'react';

export interface NavLinksContextProps {
    setNavHeight?: React.Dispatch<React.SetStateAction<number>>;
}

const NavLinksContext: Context<NavLinksContextProps> = createContext(null);

export default NavLinksContext;
