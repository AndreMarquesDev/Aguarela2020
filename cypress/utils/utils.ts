import { Locale } from '../../types/Locale';
import {
    menuIconButtonDataTestId,
    nukaCarouselNextButtonDataTestId,
} from '../../utils/dataTestIds';
import { TextsInterface, textsPt, textsEn } from '../../utils/texts';

export const getLocalizedTexts = (locale: Locale): TextsInterface => {
    if (locale === Locale.Pt) {
        return textsPt;
    }

    return textsEn;
};

export const openMenuMobile = (isMobile: boolean, isFirefox: boolean): void => {
    if (isMobile) {
        cy.getByDataTestId(menuIconButtonDataTestId).click({ force: isFirefox });
    }
};

export const clickNextArrowButtonIfMobile = (
    isMobile: boolean,
    container: string,
    containerTitle: string
): void => {
    const isFirefox = Cypress.isBrowser('firefox');

    if (isMobile) {
        cy.getByDataTestIdParent(container, nukaCarouselNextButtonDataTestId).click({
            force: isFirefox,
        });

        cy.getByText(container, containerTitle).scrollIntoView();
    }
};
