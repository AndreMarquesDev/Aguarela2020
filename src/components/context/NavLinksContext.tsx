import type { Context } from 'react';
import { createContext } from 'react';

export type NavLinksContextProps = {
    isMenuOpen: boolean;
    toggleMenu: React.Dispatch<React.SetStateAction<boolean>>;
    setNavHeight?: React.Dispatch<React.SetStateAction<number>>;
};

// @ts-ignore
export const NavLinksContext: Context<NavLinksContextProps> = createContext(null);
