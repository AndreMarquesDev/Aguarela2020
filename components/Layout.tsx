/* eslint-disable no-console */
import React, { FC } from 'react';
import { useRouter } from 'next/router';
import GeneralStyles from '../styles/styles';
import Header from './Header';
import colors from '../styles/colors';
import LanguageButton from './LanguageButton';
import { bodyPadding } from '../styles/variables';

const Layout: FC = props => {
    const { children } = props;

    const {
        route,
        query,
    } = useRouter();

    return (
        <main className="Layout">
            <GeneralStyles />

            <Header currentRoute={route} language={query.language?.toString()} />

            {children}

            <LanguageButton language={query.language?.toString()} />

            <style jsx>
                {`
                    .Layout {
                        position: relative;
                        padding: ${bodyPadding} 0;

                        &:before {
                            content: '';
                            background: ${colors.lightGrey};
                            width: 50%;
                            height: 100%;
                            position: absolute;
                            top: 0;
                            left: 0;
                            z-index: -1;
                        }
                    }
                `}
            </style>

            {/* {console.log('%c| üîß Developed by AndreMarquesDev ‚úèÔ∏è Designed by Aguarela Digital |', 'background: #000; color: #fff;')}
            {console.log('%c| https://github.com/AndreMarquesDev |', 'background: #000; color: #fff;')}
            {console.log('%c| https://www.instagram.com/aguareladigital |', 'background: #000; color: #fff;')} */}
        </main>
    );
};

export default Layout;
