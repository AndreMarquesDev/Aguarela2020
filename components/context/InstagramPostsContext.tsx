import { createContext, Context } from 'react';
import { IInstagramPost } from '../../utils/generic';

export interface IInstagramPostsContext {
    posts: IInstagramPost[];
}

const InstagramPostsContext: Context<IInstagramPostsContext> = createContext(null);

export default InstagramPostsContext;
