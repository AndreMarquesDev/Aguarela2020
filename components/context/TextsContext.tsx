import { createContext, Context } from 'react';
import { TextsInterface, textsPt } from '../../utils/texts';

export interface TextsContextProps {
    texts: TextsInterface;
}

export const TextsContext: Context<TextsContextProps> = createContext({
    texts: textsPt,
});
