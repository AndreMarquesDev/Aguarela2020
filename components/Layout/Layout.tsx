import React, { ReactNode, useContext, useMemo } from 'react';
import { useRouter } from 'next/router';
import classNames from 'classnames';
import { Header } from '../Header/Header';
import { TextsContext, TextsContextProps } from '../context/TextsContext';
import { textsEn, textsPt } from '../../utils/texts';
import { Footer } from '../Footer/Footer';
import { Locale } from '../../utils/locales';
import { NavLinksContext } from '../context/NavLinksContext';
import { useWindowSize, Breakpoint } from '../../utils/useWindowSize';

export interface LayoutProps {
    children: ReactNode;
}

export const Layout = ({ children }: LayoutProps): JSX.Element => {
    const { route, query } = useRouter();
    const { isMenuOpen } = useContext(NavLinksContext);

    const currentLanguage = query.language?.toString() as Locale;
    const texts = currentLanguage === 'pt' ? textsPt : textsEn;

    const windowSize = useWindowSize();
    const isDesktop = windowSize.width > Breakpoint.Tablet;

    const textsContextValue = useMemo<TextsContextProps>(
        () => ({
            texts,
        }),
        [texts]
    );

    return (
        <>
            <TextsContext.Provider value={textsContextValue}>
                <main className={classNames('layout', isMenuOpen && !isDesktop && 'menuOpen')}>
                    <Header currentRoute={route} language={currentLanguage} />

                    {children}

                    <Footer language={currentLanguage} />
                </main>
            </TextsContext.Provider>

            <style jsx>
                {`
                    .layout {
                        position: relative;
                    }

                    .menuOpen {
                        position: fixed;
                        overflow: hidden;
                    }
                `}
            </style>
        </>
    );
};
