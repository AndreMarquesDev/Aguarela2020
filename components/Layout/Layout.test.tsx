import '@testing-library/jest-dom';
import React from 'react';
import { render, screen, RenderResult } from '@testing-library/react';
import * as nextRouter from 'next/router';
import { Layout, LayoutProps } from './Layout';
import { NavLinksContext } from '../context/NavLinksContext';
import { Breakpoint } from '../../utils/useWindowSize';
import { setJestWindowWidth } from '../../utils/jest/setJestWindowWidth';
import { textsEn } from '../../utils/texts';

// @ts-ignore
nextRouter.useRouter = jest.fn(() => ({
    route: '/',
    query: {
        language: 'pt',
    },
}));

const baseProps: LayoutProps = {
    children: (
        <div>
            <p>Child</p>
        </div>
    ),
};

const renderComponent = (newProps?: Partial<LayoutProps>, isMenuOpen = false): RenderResult => {
    return render(
        <NavLinksContext.Provider value={{ isMenuOpen, toggleMenu: jest.fn() }}>
            <Layout {...baseProps} {...newProps} />
        </NavLinksContext.Provider>
    );
};

describe('<Layout />', () => {
    test('renders properly', () => {
        const { container } = renderComponent();

        const childElement = screen.getByText('Child');
        const logo = screen.getByAltText('Logo');

        expect(childElement).toBeInTheDocument();
        expect(logo).toBeInTheDocument();

        expect(container).toMatchSnapshot();
    });

    test('renders properly when the menu is open and the resolution is lower than desktop', () => {
        setJestWindowWidth(Breakpoint.Mobile);

        const { container } = renderComponent({}, true);

        expect(container).toMatchSnapshot();
    });

    test('renders properly in English', () => {
        (nextRouter.useRouter as jest.Mock).mockImplementation(() => ({
            route: '/',
            query: {
                language: 'en',
            },
        }));

        const { container } = renderComponent();

        const aboutLink = screen.getByText(textsEn.about);

        expect(aboutLink).toBeInTheDocument();

        expect(container).toMatchSnapshot();
    });
});
