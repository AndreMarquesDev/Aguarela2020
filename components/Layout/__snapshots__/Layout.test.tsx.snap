// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> renders properly 1`] = `
<div>
  <main
    class="jsx-2984514796 layout"
  >
    <header
      class="jsx-2230556198"
    >
      <nav
        class="jsx-2230556198 wrapper"
      >
        <a
          class="jsx-2230556198"
          href="/pt/homepage"
        >
          <div
            style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
          >
            <div
              style="box-sizing: border-box; display: block; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                role="presentation"
                src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjUwIiBoZWlnaHQ9IjM1IiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZlcnNpb249IjEuMSIvPg=="
                style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
              />
            </div>
            <img
              alt="Logo"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="visibility: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
          </div>
        </a>
        <ul
          class="jsx-1064030420 "
        >
          <li
            class="jsx-1064030420"
          >
            <a
              class="jsx-1064030420 "
              href="/pt/about"
            >
              Sobre
            </a>
          </li>
          <li
            class="jsx-1064030420"
          >
            <a
              class="jsx-1064030420 "
              href="/pt/projects"
            >
              Projetos
            </a>
          </li>
          <li
            class="jsx-1064030420"
          >
            <a
              class="jsx-1064030420 "
              href="/pt/services"
            >
              Serviços
            </a>
          </li>
          <li
            class="jsx-1064030420"
          >
            <a
              class="jsx-1064030420 "
              href="/pt/contact"
            >
              Contacto
            </a>
          </li>
          <li
            class="jsx-1064030420 languageButton"
          >
            <button
              class="jsx-1698987686"
              type="button"
            >
              pt
            </button>
          </li>
        </ul>
        <i
          aria-label="toggle menu"
          class="jsx-2133906697 menuIcon hidden"
          role="button"
          tabindex="0"
        >
          <span
            class="jsx-2133906697"
          />
        </i>
      </nav>
    </header>
    <div>
      <p>
        Child
      </p>
    </div>
    <footer
      class="jsx-1577276803"
      data-testid="footer_section"
    >
      <div
        class="jsx-1577276803 wrapper"
      >
        <p
          class="jsx-1577276803"
        >
          Aguarela Digital - Developed by
           
          <a
            class="jsx-1577276803 animatedLink animatedLinkWhite"
            href="https://www.andremarquesdev.com"
            rel="noreferrer"
            target="_blank"
          >
            André Marques
          </a>
        </p>
        <ul
          class="jsx-1577276803"
        >
          <li
            class="jsx-1577276803"
          >
            <a
              aria-label="Aguarela instagram"
              class="jsx-1577276803"
              href="https://www.instagram.com/aguareladigital"
              rel="noreferrer"
              target="_blank"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 448 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M224.1 141c-63.6 0-114.9 51.3-114.9 114.9s51.3 114.9 114.9 114.9S339 319.5 339 255.9 287.7 141 224.1 141zm0 189.6c-41.1 0-74.7-33.5-74.7-74.7s33.5-74.7 74.7-74.7 74.7 33.5 74.7 74.7-33.6 74.7-74.7 74.7zm146.4-194.3c0 14.9-12 26.8-26.8 26.8-14.9 0-26.8-12-26.8-26.8s12-26.8 26.8-26.8 26.8 12 26.8 26.8zm76.1 27.2c-1.7-35.9-9.9-67.7-36.2-93.9-26.2-26.2-58-34.4-93.9-36.2-37-2.1-147.9-2.1-184.9 0-35.8 1.7-67.6 9.9-93.9 36.1s-34.4 58-36.2 93.9c-2.1 37-2.1 147.9 0 184.9 1.7 35.9 9.9 67.7 36.2 93.9s58 34.4 93.9 36.2c37 2.1 147.9 2.1 184.9 0 35.9-1.7 67.7-9.9 93.9-36.2 26.2-26.2 34.4-58 36.2-93.9 2.1-37 2.1-147.8 0-184.8zM398.8 388c-7.8 19.6-22.9 34.7-42.6 42.6-29.5 11.7-99.5 9-132.1 9s-102.7 2.6-132.1-9c-19.6-7.8-34.7-22.9-42.6-42.6-11.7-29.5-9-99.5-9-132.1s-2.6-102.7 9-132.1c7.8-19.6 22.9-34.7 42.6-42.6 29.5-11.7 99.5-9 132.1-9s102.7-2.6 132.1 9c19.6 7.8 34.7 22.9 42.6 42.6 11.7 29.5 9 99.5 9 132.1s2.7 102.7-9 132.1z"
                />
              </svg>
            </a>
          </li>
          <li
            class="jsx-1577276803"
          >
            <a
              aria-label="Aguarela facebook"
              class="jsx-1577276803"
              href="https://www.facebook.com/aguareladigitalagency"
              rel="noreferrer"
              target="_blank"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 320 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M279.14 288l14.22-92.66h-88.91v-60.13c0-25.35 12.42-50.06 52.24-50.06h40.42V6.26S260.43 0 225.36 0c-73.22 0-121.08 44.38-121.08 124.72v70.62H22.89V288h81.39v224h100.17V288z"
                />
              </svg>
            </a>
          </li>
          <li
            class="jsx-1577276803"
          >
            <a
              aria-label="Go to contact page"
              class="jsx-1577276803"
              href="/pt/contact"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M502.3 190.8c3.9-3.1 9.7-.2 9.7 4.7V400c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V195.6c0-5 5.7-7.8 9.7-4.7 22.4 17.4 52.1 39.5 154.1 113.6 21.1 15.4 56.7 47.8 92.2 47.6 35.7.3 72-32.8 92.3-47.6 102-74.1 131.6-96.3 154-113.7zM256 320c23.2.4 56.6-29.2 73.4-41.4 132.7-96.3 142.8-104.7 173.4-128.7 5.8-4.5 9.2-11.5 9.2-18.9v-19c0-26.5-21.5-48-48-48H48C21.5 64 0 85.5 0 112v19c0 7.4 3.4 14.3 9.2 18.9 30.6 23.9 40.7 32.4 173.4 128.7 16.8 12.2 50.2 41.8 73.4 41.4z"
                />
              </svg>
            </a>
          </li>
        </ul>
      </div>
    </footer>
  </main>
</div>
`;

exports[`<Layout /> renders properly in English 1`] = `
<div>
  <main
    class="jsx-2984514796 layout"
  >
    <header
      class="jsx-2230556198"
    >
      <nav
        class="jsx-2230556198 wrapper"
      >
        <a
          class="jsx-2230556198"
          href="/en/homepage"
        >
          <div
            style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
          >
            <div
              style="box-sizing: border-box; display: block; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                role="presentation"
                src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTUwIiBoZWlnaHQ9IjM1IiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZlcnNpb249IjEuMSIvPg=="
                style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
              />
            </div>
            <img
              alt="Logo"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="visibility: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
          </div>
        </a>
        <ul
          class="jsx-1064030420 mobileLayout"
        >
          <li
            class="jsx-1064030420"
          >
            <a
              class="jsx-1064030420 "
              href="/en/about"
            >
              About
            </a>
          </li>
          <li
            class="jsx-1064030420"
          >
            <a
              class="jsx-1064030420 "
              href="/en/projects"
            >
              Projects
            </a>
          </li>
          <li
            class="jsx-1064030420"
          >
            <a
              class="jsx-1064030420 "
              href="/en/services"
            >
              Services
            </a>
          </li>
          <li
            class="jsx-1064030420"
          >
            <a
              class="jsx-1064030420 "
              href="/en/contact"
            >
              Contact
            </a>
          </li>
          <li
            class="jsx-1064030420 languageButton white"
          >
            <button
              class="jsx-1698987686"
              type="button"
            >
              pt
            </button>
          </li>
        </ul>
        <i
          aria-label="toggle menu"
          class="jsx-2133906697 menuIcon"
          role="button"
          tabindex="0"
        >
          <span
            class="jsx-2133906697"
          />
        </i>
      </nav>
    </header>
    <div>
      <p>
        Child
      </p>
    </div>
    <footer
      class="jsx-1577276803"
      data-testid="footer_section"
    >
      <div
        class="jsx-1577276803 wrapper"
      >
        <p
          class="jsx-1577276803"
        >
          Aguarela Digital - Developed by
           
          <a
            class="jsx-1577276803 animatedLink animatedLinkWhite"
            href="https://www.andremarquesdev.com"
            rel="noreferrer"
            target="_blank"
          >
            André Marques
          </a>
        </p>
        <ul
          class="jsx-1577276803"
        >
          <li
            class="jsx-1577276803"
          >
            <a
              aria-label="Aguarela instagram"
              class="jsx-1577276803"
              href="https://www.instagram.com/aguareladigital"
              rel="noreferrer"
              target="_blank"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 448 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M224.1 141c-63.6 0-114.9 51.3-114.9 114.9s51.3 114.9 114.9 114.9S339 319.5 339 255.9 287.7 141 224.1 141zm0 189.6c-41.1 0-74.7-33.5-74.7-74.7s33.5-74.7 74.7-74.7 74.7 33.5 74.7 74.7-33.6 74.7-74.7 74.7zm146.4-194.3c0 14.9-12 26.8-26.8 26.8-14.9 0-26.8-12-26.8-26.8s12-26.8 26.8-26.8 26.8 12 26.8 26.8zm76.1 27.2c-1.7-35.9-9.9-67.7-36.2-93.9-26.2-26.2-58-34.4-93.9-36.2-37-2.1-147.9-2.1-184.9 0-35.8 1.7-67.6 9.9-93.9 36.1s-34.4 58-36.2 93.9c-2.1 37-2.1 147.9 0 184.9 1.7 35.9 9.9 67.7 36.2 93.9s58 34.4 93.9 36.2c37 2.1 147.9 2.1 184.9 0 35.9-1.7 67.7-9.9 93.9-36.2 26.2-26.2 34.4-58 36.2-93.9 2.1-37 2.1-147.8 0-184.8zM398.8 388c-7.8 19.6-22.9 34.7-42.6 42.6-29.5 11.7-99.5 9-132.1 9s-102.7 2.6-132.1-9c-19.6-7.8-34.7-22.9-42.6-42.6-11.7-29.5-9-99.5-9-132.1s-2.6-102.7 9-132.1c7.8-19.6 22.9-34.7 42.6-42.6 29.5-11.7 99.5-9 132.1-9s102.7-2.6 132.1 9c19.6 7.8 34.7 22.9 42.6 42.6 11.7 29.5 9 99.5 9 132.1s2.7 102.7-9 132.1z"
                />
              </svg>
            </a>
          </li>
          <li
            class="jsx-1577276803"
          >
            <a
              aria-label="Aguarela facebook"
              class="jsx-1577276803"
              href="https://www.facebook.com/aguareladigitalagency"
              rel="noreferrer"
              target="_blank"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 320 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M279.14 288l14.22-92.66h-88.91v-60.13c0-25.35 12.42-50.06 52.24-50.06h40.42V6.26S260.43 0 225.36 0c-73.22 0-121.08 44.38-121.08 124.72v70.62H22.89V288h81.39v224h100.17V288z"
                />
              </svg>
            </a>
          </li>
          <li
            class="jsx-1577276803"
          >
            <a
              aria-label="Go to contact page"
              class="jsx-1577276803"
              href="/en/contact"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M502.3 190.8c3.9-3.1 9.7-.2 9.7 4.7V400c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V195.6c0-5 5.7-7.8 9.7-4.7 22.4 17.4 52.1 39.5 154.1 113.6 21.1 15.4 56.7 47.8 92.2 47.6 35.7.3 72-32.8 92.3-47.6 102-74.1 131.6-96.3 154-113.7zM256 320c23.2.4 56.6-29.2 73.4-41.4 132.7-96.3 142.8-104.7 173.4-128.7 5.8-4.5 9.2-11.5 9.2-18.9v-19c0-26.5-21.5-48-48-48H48C21.5 64 0 85.5 0 112v19c0 7.4 3.4 14.3 9.2 18.9 30.6 23.9 40.7 32.4 173.4 128.7 16.8 12.2 50.2 41.8 73.4 41.4z"
                />
              </svg>
            </a>
          </li>
        </ul>
      </div>
    </footer>
  </main>
</div>
`;

exports[`<Layout /> renders properly when the menu is open and the resolution is lower than desktop 1`] = `
<div>
  <main
    class="jsx-2984514796 layout menuOpen"
  >
    <header
      class="jsx-2230556198"
    >
      <nav
        class="jsx-2230556198 wrapper"
      >
        <a
          class="jsx-2230556198"
          href="/pt/homepage"
        >
          <div
            style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
          >
            <div
              style="box-sizing: border-box; display: block; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                role="presentation"
                src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTUwIiBoZWlnaHQ9IjM1IiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZlcnNpb249IjEuMSIvPg=="
                style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
              />
            </div>
            <img
              alt="Logo"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="visibility: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
          </div>
        </a>
        <ul
          class="jsx-1064030420 mobileLayout menuOpen"
        >
          <li
            class="jsx-1064030420"
          >
            <a
              class="jsx-1064030420 "
              href="/pt/about"
            >
              Sobre
            </a>
          </li>
          <li
            class="jsx-1064030420"
          >
            <a
              class="jsx-1064030420 "
              href="/pt/projects"
            >
              Projetos
            </a>
          </li>
          <li
            class="jsx-1064030420"
          >
            <a
              class="jsx-1064030420 "
              href="/pt/services"
            >
              Serviços
            </a>
          </li>
          <li
            class="jsx-1064030420"
          >
            <a
              class="jsx-1064030420 "
              href="/pt/contact"
            >
              Contacto
            </a>
          </li>
          <li
            class="jsx-1064030420 languageButton white"
          >
            <button
              class="jsx-1698987686"
              type="button"
            >
              pt
            </button>
          </li>
        </ul>
        <i
          aria-label="toggle menu"
          class="jsx-2133906697 menuIcon open"
          role="button"
          tabindex="0"
        >
          <span
            class="jsx-2133906697"
          />
        </i>
      </nav>
    </header>
    <div>
      <p>
        Child
      </p>
    </div>
    <footer
      class="jsx-1577276803"
      data-testid="footer_section"
    >
      <div
        class="jsx-1577276803 wrapper"
      >
        <p
          class="jsx-1577276803"
        >
          Aguarela Digital - Developed by
           
          <a
            class="jsx-1577276803 animatedLink animatedLinkWhite"
            href="https://www.andremarquesdev.com"
            rel="noreferrer"
            target="_blank"
          >
            André Marques
          </a>
        </p>
        <ul
          class="jsx-1577276803"
        >
          <li
            class="jsx-1577276803"
          >
            <a
              aria-label="Aguarela instagram"
              class="jsx-1577276803"
              href="https://www.instagram.com/aguareladigital"
              rel="noreferrer"
              target="_blank"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 448 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M224.1 141c-63.6 0-114.9 51.3-114.9 114.9s51.3 114.9 114.9 114.9S339 319.5 339 255.9 287.7 141 224.1 141zm0 189.6c-41.1 0-74.7-33.5-74.7-74.7s33.5-74.7 74.7-74.7 74.7 33.5 74.7 74.7-33.6 74.7-74.7 74.7zm146.4-194.3c0 14.9-12 26.8-26.8 26.8-14.9 0-26.8-12-26.8-26.8s12-26.8 26.8-26.8 26.8 12 26.8 26.8zm76.1 27.2c-1.7-35.9-9.9-67.7-36.2-93.9-26.2-26.2-58-34.4-93.9-36.2-37-2.1-147.9-2.1-184.9 0-35.8 1.7-67.6 9.9-93.9 36.1s-34.4 58-36.2 93.9c-2.1 37-2.1 147.9 0 184.9 1.7 35.9 9.9 67.7 36.2 93.9s58 34.4 93.9 36.2c37 2.1 147.9 2.1 184.9 0 35.9-1.7 67.7-9.9 93.9-36.2 26.2-26.2 34.4-58 36.2-93.9 2.1-37 2.1-147.8 0-184.8zM398.8 388c-7.8 19.6-22.9 34.7-42.6 42.6-29.5 11.7-99.5 9-132.1 9s-102.7 2.6-132.1-9c-19.6-7.8-34.7-22.9-42.6-42.6-11.7-29.5-9-99.5-9-132.1s-2.6-102.7 9-132.1c7.8-19.6 22.9-34.7 42.6-42.6 29.5-11.7 99.5-9 132.1-9s102.7-2.6 132.1 9c19.6 7.8 34.7 22.9 42.6 42.6 11.7 29.5 9 99.5 9 132.1s2.7 102.7-9 132.1z"
                />
              </svg>
            </a>
          </li>
          <li
            class="jsx-1577276803"
          >
            <a
              aria-label="Aguarela facebook"
              class="jsx-1577276803"
              href="https://www.facebook.com/aguareladigitalagency"
              rel="noreferrer"
              target="_blank"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 320 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M279.14 288l14.22-92.66h-88.91v-60.13c0-25.35 12.42-50.06 52.24-50.06h40.42V6.26S260.43 0 225.36 0c-73.22 0-121.08 44.38-121.08 124.72v70.62H22.89V288h81.39v224h100.17V288z"
                />
              </svg>
            </a>
          </li>
          <li
            class="jsx-1577276803"
          >
            <a
              aria-label="Go to contact page"
              class="jsx-1577276803"
              href="/pt/contact"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M502.3 190.8c3.9-3.1 9.7-.2 9.7 4.7V400c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V195.6c0-5 5.7-7.8 9.7-4.7 22.4 17.4 52.1 39.5 154.1 113.6 21.1 15.4 56.7 47.8 92.2 47.6 35.7.3 72-32.8 92.3-47.6 102-74.1 131.6-96.3 154-113.7zM256 320c23.2.4 56.6-29.2 73.4-41.4 132.7-96.3 142.8-104.7 173.4-128.7 5.8-4.5 9.2-11.5 9.2-18.9v-19c0-26.5-21.5-48-48-48H48C21.5 64 0 85.5 0 112v19c0 7.4 3.4 14.3 9.2 18.9 30.6 23.9 40.7 32.4 173.4 128.7 16.8 12.2 50.2 41.8 73.4 41.4z"
                />
              </svg>
            </a>
          </li>
        </ul>
      </div>
    </footer>
  </main>
</div>
`;
